packages/collector/src/announceCycle/agentready.js

```diff
@@ -247,7 +253,7 @@ function sendEOLEvent() {
  * support for events.)
  */
 function detectEOLNodeVersion() {
-  if (isNodeVersionEOL()) {
+  if (isNodeVersionEOL() && !disableEOLEvents) {
```

Comment:
There is no need to call and process `isNodeVersionEOL()`. if its disabled.

Please put the check to line 118.